src/controllers/resources.controller.ts

```diff
@@ -1,36 +1,35 @@
-import { WithId, Document } from "mongodb";
-import { convertResultToStudyPlan, convertResultToResource } from "@utils/converter";
+import { Document, WithId } from "mongodb";
+import { convertResultToQuickLink, convertResultToResource, convertResultToStudyPlan } from "@utils/converter";
 import { getConnection } from "@utils/mongoconnection";
 import { Request, Response } from "express";
 
+const converters = new Map<string, (result: WithId<Document>) => any>([
+	["study_plans", convertResultToStudyPlan],
+	["resources", convertResultToResource],
+	["quicklinks", convertResultToQuickLink],
+]);
```

Comment:
...?

```typescript
    const converters = {
      "study_plans": convertResultToStudyPlan,
      "resources": convertResultToResource,
      "quicklinks": convertResultToQuickLink,
    };
```

?

And in the wrong page handling we can do stuff like
```typescript
    if (!converterFunc) {
      res.status(400).send(`The page ${pageName} is not supported. Supported pages are: ${Object.keys(converters).join(', ')}`);
      return;
    }
```

Reply:
I swear I did this before and it didn't work. I'll try again

Reply:
Hmmmm...